### PR TITLE
Fix `_exists_` in query_string on empty indices.

### LINKED ISSUE #25956: NPE when searching * against an index with default field set (but field does not exist)
ES 5.5

When an index has a default field defined in the index settings

```
"settings": {
   "index": {
    "query": {
     "default_field": "message"
    },
    "number_of_replicas": "1"
   }
  }
```

but the actual field does not exist, a * query will cause a NPE:

```
{
  "error": {
    "root_cause": [
      {
        "type": "query_shard_exception",
        "reason": "failed to create query: {\n  \"bool\" : {\n    \"must\" : [\n      {\n        \"query_string\" : {\n          \"query\" : \"*\",\n          \"fields\" : [ ],\n          \"use_dis_max\" : true,\n          \"tie_breaker\" : 0.0,\n          \"default_operator\" : \"or\",\n          \"auto_generate_phrase_queries\" : false,\n          \"max_determinized_states\" : 10000,\n          \"enable_position_increments\" : true,\n          \"fuzziness\" : \"AUTO\",\n          \"fuzzy_prefix_length\" : 0,\n          \"fuzzy_max_expansions\" : 50,\n          \"phrase_slop\" : 0,\n          \"escape\" : false,\n          \"split_on_whitespace\" : true,\n          \"boost\" : 1.0\n        }\n      }\n    ],\n    \"disable_coord\" : false,\n    \"adjust_pure_negative\" : true,\n    \"boost\" : 1.0\n  }\n}",
        "index_uuid": "SRYVdIE4R5mF0kHJgHBZVw",
        "index": "logstash-test"
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "query",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "logstash-test",
        "node": "68R5f1xiTwyxWmchtjr1Ww",
        "reason": {
          "type": "query_shard_exception",
          "reason": "failed to create query: {\n  \"bool\" : {\n    \"must\" : [\n      {\n        \"query_string\" : {\n          \"query\" : \"*\",\n          \"fields\" : [ ],\n          \"use_dis_max\" : true,\n          \"tie_breaker\" : 0.0,\n          \"default_operator\" : \"or\",\n          \"auto_generate_phrase_queries\" : false,\n          \"max_determinized_states\" : 10000,\n          \"enable_position_increments\" : true,\n          \"fuzziness\" : \"AUTO\",\n          \"fuzzy_prefix_length\" : 0,\n          \"fuzzy_max_expansions\" : 50,\n          \"phrase_slop\" : 0,\n          \"escape\" : false,\n          \"split_on_whitespace\" : true,\n          \"boost\" : 1.0\n        }\n      }\n    ],\n    \"disable_coord\" : false,\n    \"adjust_pure_negative\" : true,\n    \"boost\" : 1.0\n  }\n}",
          "index_uuid": "SRYVdIE4R5mF0kHJgHBZVw",
          "index": "logstash-test",
          "caused_by": {
            "type": "null_pointer_exception",
            "reason": null
          }
        }
      }
    ]
  },
  "status": 400
}
```

See repro and log snippet [here](https://gist.github.com/ppf2/d58a318ee58a06f7e2b58473758fb25c).

### PR DESCRIPTION
**Pull request by jpountz**
It currently fails if there are no mappings yet.

Closes #25956

### CODE REVIEW

**Comment by jpountz** [comment]
@elasticmachine test it again

**Comment by jpountz** [comment]
@elasticmachine please test it again

File: core/src/test/java/org/elasticsearch/index/query/QueryStringQueryBuilderTests.java [hunk 1]
```diff
@@ -462,6 +462,10 @@ public void testToQueryWithGraph() throws Exception {
         }
     }
 
+    public void testExistsQuery() {
+        
+    }
```
Comment (jimczi) [inline core/src/test/java/org/elasticsearch/index/query/QueryStringQueryBuilderTests.java]: nit: leftover

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**